### PR TITLE
Add `finalized` metadata field to API responses

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Implements https://github.com/ethereum/beacon-APIs/pull/254 for the following endpoints:
- [ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/getBlockAttestations](https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/getBlockAttestations)
- [ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/getBlockHeaders](https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/getBlockHeaders)
- [ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/getBlockHeader](https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/getBlockHeader)
- [ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/getBlockRoot](https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/getBlockRoot)

It also simplifies some tests as they had unnecessary setup.

**Which issues(s) does this PR fix?**

Part of #11747

### CODE REVIEW

File: beacon-chain/rpc/eth/beacon/blocks.go [hunk 1]
```diff
@@ -173,6 +174,7 @@ func (bs *Server) ListBlockHeaders(ctx context.Context, req *ethpbv1.BlockHeader
 	}
 
 	isOptimistic := false
+	isFinalized := true
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/blocks.go:177]: We assume finalized and as long as we "are finalized", we update the value inside the loop.

File: beacon-chain/rpc/eth/beacon/blocks_test.go [hunk 2]
```diff
@@ -387,100 +417,105 @@ func TestServer_ListBlockHeaders(t *testing.T) {
...
 	util.SaveBlock(t, ctx, beaconDB, b3)
 	b4 := util.NewBeaconBlock()
-	b4.Block.Slot = 31
+	b4.Block.Slot = 28
 	b4.Block.ParentRoot = bytesutil.PadTo([]byte{1}, 32)
 	util.SaveBlock(t, ctx, beaconDB, b4)
-	b5 := util.NewBeaconBlock()
-	b5.Block.Slot = 28
-	b5.Block.ParentRoot = bytesutil.PadTo([]byte{1}, 32)
-	util.SaveBlock(t, ctx, beaconDB, b5)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/blocks_test.go:420]: This diff is misleading, I simply changed names of `b2/b3/b4/b5` to `b1/b2/b3/b4`.

**Review by saolyn — APPROVED** [review]
lgtm

**Review by rauljordan — DISMISSED** [review]

**Review by saolyn — APPROVED** [review]
if tests now pass, lgtm

**Pull request merged**